ice40/CMakeLists.txt

```diff
@@ -0,0 +1,88 @@
+cmake_minimum_required(VERSION 3.5)
+project(chipdb-ice40 NONE)
+
+set(ALL_ICE40_DEVICES 384 1k 5k u4k 8k)
+set(ICE40_DEVICES ${ALL_ICE40_DEVICES} CACHE STRING
+    "Include support for these iCE40 devices (available: ${ALL_ICE40_DEVICES})")
+message(STATUS "Enabled iCE40 devices: ${ICE40_DEVICES}")
+
+if(DEFINED ICE40_CHIPDB)
+    add_custom_target(chipdb-ice40-bbas ALL)
+else()
+    find_package(PythonInterp 3.5 REQUIRED)
+
+    # shared among all families
+    set(SERIALIZE_CHIPDBS TRUE CACHE BOOL
+        "Serialize device data preprocessing to minimize memory use")
+
+    set(icestorm_default_install_prefix ${CMAKE_INSTALL_PREFIX})
+    # for compatibility with old build scripts
+    if(DEFINED ICEBOX_ROOT)
+        message(WARNING "-DICEBOX_ROOT= is deprecated, use -DICESTORM_INSTALL_PREFIX= instead")
+        get_filename_component(dir ${ICEBOX_ROOT} DIRECTORY)
+        get_filename_component(dir ${dir} DIRECTORY)
+        set(icestorm_default_install_prefix ${dir})
+    endif()
+    set(ICESTORM_INSTALL_PREFIX ${icestorm_default_install_prefix} CACHE STRING
+        "IceStorm install prefix")
+    message(STATUS "IceStorm install prefix: ${ICESTORM_INSTALL_PREFIX}")
+
+    if(NOT DEFINED ICEBOX_DATADIR)
+        set(ICEBOX_DATADIR ${ICESTORM_INSTALL_PREFIX}/share/icebox)
+    endif()
+    message(STATUS "icebox data directory: ${ICEBOX_DATADIR}")
+
+    set(all_device_bbas)
+    file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/chipdb)
+    foreach(device ${ICE40_DEVICES})
+        if(NOT device IN_LIST ALL_ICE40_DEVICES)
+            message(FATAL_ERROR "Device ${device} is not a supported iCE40 device")
+        endif()
+
+        if(device STREQUAL "5k")
+            set(timing_opts
+                --slow ${ICEBOX_DATADIR}/timings_up5k.txt)
+        elseif(device STREQUAL "u4k")
+            set(timing_opts
+                --slow ${ICEBOX_DATADIR}/timings_u4k.txt)
+        elseif(device STREQUAL "384")
+            set(timing_opts
+                --slow ${ICEBOX_DATADIR}/timings_lp384.txt)
+        else()
+            set(timing_opts
+                --fast ${ICEBOX_DATADIR}/timings_hx${device}.txt
+                --slow ${ICEBOX_DATADIR}/timings_lp${device}.txt)
+        endif()
+
+        set(device_bba chipdb/chipdb-${device}.bba)
+        add_custom_command(
+            OUTPUT ${device_bba}
+            COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/chipdb.py
+                -p ${CMAKE_CURRENT_SOURCE_DIR}/constids.inc
+                -g ${CMAKE_CURRENT_SOURCE_DIR}/gfx.h
+                ${timing_opts}
+                ${ICEBOX_DATADIR}/chipdb-${device}.txt
+                > ${device_bba}
```

Comment:
it's a bit of a shame to lose the `mv` step that the old rules had, as if generation fails mid-way (e.g. OOM killed) then something will now fail more subtly on a half-written bba file rather than just re-running the generator script.

Reply:
Ah I missed this, added back but without a dependency on Unix `mv`